src/main/java/com/zeroclue/jmeter/protocol/amqp/AMQPConsumer.java

```diff
@@ -72,11 +84,12 @@ public SampleResult sample(Entry entry) {
             // only do this once per thread, otherwise it slows down the consumption by appx 50%
             if (consumer == null) {
                 log.info("Creating consumer");
-                consumer = new QueueingConsumer(channel);
+                response = new LinkedBlockingQueue<>(1);
+                consumer = (consumerTag, delivery) -> response.offer(delivery);
             }
             if (consumerTag == null) {
                 log.info("Starting basic consumer");
-                consumerTag = channel.basicConsume(getQueue(), autoAck(), consumer);
+                consumerTag = channel.basicConsume(getQueue(), autoAck(), consumer, consumerTag  -> {});
```

Comment:
This is the only part, I'm not completely sure about if the `CancelCallback` can simply be ignored.